### PR TITLE
Sticker Packs

### PR DESCRIPTION
**Pull request by Sjmarf**
> **Warning**
> This relies on DiscordKit being updated to include [my recent PR](https://github.com/SwiftcordApp/DiscordKit/pull/27).

- Changed Stickers to use Buttons for click detection instead of `.onTapGesture`. This means that the Bird sticker, which previously wouldn't show a preview when clicked, now works correctly.
- Added a Sticker Pack view that can be accessed from the sticker preview view. This can be a little laggy on the larger sticker packs, but it's usable (on my device).
- Moved the code for hover detection from the sticker-view used in messages to the basic `StickerItemView`. This is because the hover-detection code is reused in the sticker pack view.
- Renamed `StickerView` (the view used to display stickers in messages) to `MessageStickerView`.

### CODE REVIEW

File: Swiftcord.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -532,6 +535,7 @@
 		DA4A887B27C0AF3000720909 = {
 			isa = PBXGroup;
 			children = (
+				03A188B82A0F864500D5F4BC /* DiscordKit */,
```
Comment (cryptoAlgorithm) [inline Swiftcord.xcodeproj/project.pbxproj:538]:
```suggestion
```

Similarly, please ensure local references to DiscordKit aren't committed.

File: Swiftcord/Views/Message/MessageStickerView.swift [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+//  Created by Vincent Kwok on 23/2/22.
+//
+
+import SwiftUI
+import Lottie
+import DiscordKitCore
+import DiscordKit
+import CachedAsyncImage
+
+
```
Comment (cryptoAlgorithm) [inline Swiftcord/Views/Message/MessageStickerView.swift:14]: There are too many blank lines here, please ensure there's only one blank line at most to pass the linting rule.

File: Swiftcord/Views/Message/MessageStickerView.swift [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+			Task {
+				   fullSticker = try await restAPI.getSticker(sticker.id)
+				   openPopoverEvt()
+			   }
+		   } else {
+			   openPopoverEvt()
+		   }
+		   infoShow.toggle()
+		   packPresenting = false
+
```
Comment (cryptoAlgorithm) [inline Swiftcord/Views/Message/MessageStickerView.swift:144]:
```suggestion
```

For consistency, please remove this blank line before the closing brace.

**Review by cryptoAlgorithm — CHANGES_REQUESTED** [review]
Overall, this PR was high quality, but there were some issues with changes to the project file and some other lint warnings. Applying all suggested fixes should resolve issues to the project file, but lint errors need to be manually resolved. Swiftcord's codebase has switched to only using spaces for indentation, so I suggest the same be done here to prevent mixed indentation warnings.

**Comment by cryptoAlgorithm** [comment]
I'd have preferred if all suggested changes were batched into a single commit, but I guess this works too as it would be eventually squashed for merging.

**Comment by Sjmarf** [comment]
It should be good now, I think 👍

File: Swiftcord.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -263,6 +263,9 @@
 /* End PBXBuildFile section */
 
 /* Begin PBXFileReference section */
+		03A188B82A0F864500D5F4BC /* DiscordKit */ = {isa = PBXFileReference; lastKnownFileType = wrapper; name = DiscordKit; path = ../DiscordKit; sourceTree = "<group>"; };
```
[the remark this answers is not in the document]
Comment (cryptoAlgorithm) [reply Swiftcord.xcodeproj/project.pbxproj:266]: Looks like there are still remnants of references to a local copy of DiscordKit - I recommend simply removing the reference from your local copy before committing, as the counterpart PR has already been merged on the DiscordKit side. You'll also have to change the DiscordKit reference from a commit to the `main` branch as follows:

Screenshot 2023-05-14 at 10 02 59 PM

**Review by cryptoAlgorithm — CHANGES_REQUESTED** [review]

**Comment by cryptoAlgorithm** [comment]
> It should be good now, I think 👍

Nice, there doesn't appear to be any more lint warnings/errors remaining! However, there still are some references to a local copy of DiscordKit, which shouldn't be committed.

**Comment by cryptoAlgorithm** [comment]
Ok, one _final_ issue: looks like the commit hash of DiscordKit in `Package.resolved` is still pointing to an old commit, it needs to point to at least the commit that adds the required symbols, else building on CI will not succeed. You can follow these instructions https://github.com/SwiftcordApp/Swiftcord/pull/150#discussion_r1193153536 to update the dependancy reference.

**Comment by Sjmarf** [comment]
> Ok, one _final_ issue: looks like the commit hash of DiscordKit in `Package.resolved` is still pointing to an old commit, it needs to point to at least the commit that adds the required symbols, else building on CI will not succeed. You can follow these instructions [#150 (comment)](https://github.com/SwiftcordApp/Swiftcord/pull/150#discussion_r1193153536) to update the dependancy reference.

Hopefully it's working now 😅

**Pull request merged**